src/BinSkim.Driver/RoslynAnalyzer/ILDiagnosticsAnalyzer.cs

```diff
@@ -0,0 +1,110 @@
+
+        private ILDiagnosticsAnalyzer(RoslynAnalysisContext analysisContext)
+        {
+            GlobalRoslynAnalysisContext = analysisContext;
+        }
+
+        public RoslynAnalysisContext GlobalRoslynAnalysisContext { get; private set; }
+
+        public static ILDiagnosticsAnalyzer Create(params string[] analyzerFilePaths)
+        {
+            return Create((IEnumerable<string>)analyzerFilePaths);
+        }
+
+        public static ILDiagnosticsAnalyzer Create(IEnumerable<string> analyzerFilePaths)
+        {
+            var analysisContext = new RoslynAnalysisContext();
+
+            foreach(string analyzerFilePath in analyzerFilePaths)
+            {
+                LoadAnalyzer(analyzerFilePath, analysisContext);
+            }
+
+            return Create(analysisContext);
+        }
+
+        public static ILDiagnosticsAnalyzer Create(RoslynAnalysisContext analysisContext)
+        {
+            return new ILDiagnosticsAnalyzer(analysisContext);
+        }
+
+        public static RoslynAnalysisContext LoadAnalyzer(string path, RoslynAnalysisContext analysisContext = null)
+        {
+            analysisContext = analysisContext ?? new RoslynAnalysisContext();
+
+            Assembly assembly = Assembly.LoadFrom(path);
+
+            foreach (Type type in assembly.GetTypes())
+            {
+                if (typeof(DiagnosticAnalyzer).IsAssignableFrom(type) && !type.IsAbstract)
+                {
+                    var analyzer = (DiagnosticAnalyzer)Activator.CreateInstance(type);
+                    analyzer.Initialize(analysisContext);
+                }
+            }
+            return analysisContext;
+        }
+
+        public void Analyze(string targetPath, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Create a Roslyn representation of the IL by constructing a MetadataReference against
+            // the target path (as if we intended to reference this binary during compilation, instead
+            // of analyzing it). Using this mechanism, we can scan types/members contained in the 
+            // binary. We cannot currently retrieve IL from method bodies.
+            var reference = MetadataReference.CreateFromFile(targetPath);
+            var compilation = CSharpCompilation.Create("_", references: new[] { reference });
+            var target = compilation.GetAssemblyOrModuleSymbol(reference);
+
+            // For each analysis target, we create a compilation start context, which may result
+            // in symbol action registration. We need to capture and throw these registrations 
+            // away for each binary we inspect. 
+            var compilationStartAnalysisContext = new RoslynCompilationStartAnalysisContext(compilation, _options, cancellationToken);
+            _perCompilationSymbolsActions = compilationStartAnalysisContext.SymbolActions;
```

Comment:
I would pass these down to `Analyze(ISymbol, ...)` and avoid the field that we have to null out. It will get captured with the right lifetime in the `symbol => Analyze(...)` closure.